lib/improver/database/spotdata_database.py

```diff
@@ -0,0 +1,276 @@
+import pandas as pd
+import sqlite3
+import os
+from datetime import datetime as dt
+from pandas import DataFrame
+
+
+# For debugging
+from profiler import profile
+import pprint
+import resource
+
+
+class SpotDatabase(object):
+
+    def __init__(self, cubelist,
+                 primary_dim='time',
+                 primary_map=['validity_date', 'validity_time'],
+                 primary_func=[lambda x:dt.utcfromtimestamp(x).date(),
+                               lambda x:dt.utcfromtimestamp(x).hour*100],
+
+                 pivot_dim='forecast_period',
+                 pivot_map=lambda x: 'fcr_tplus{:03d}'.format(int(x/3600)),
+
+                 column_dims=['wmo_site', 'name'],
+                 column_maps=['station_id', 'cf_name']):
+
+        """
+        Initialise class.
+
+        """
+
+        self.cubelist = cubelist
+        self.pivot_dim = pivot_dim
+        self.pivot_map = pivot_map
+        self.primary_dim = primary_dim
+        self.primary_map = primary_map
+        self.primary_func = primary_func
+
+        self.column_dims = column_dims
+        self.column_maps = column_maps
+        self.column_func = []
+
+        self.assert_similar()
+
+    def __repr__(self):
+        """
+        Representation of the instance.
+
+        """
+
+        return '<SpotDatabase: {}>'.format(self.primary_dim)
+
+    def assert_similar(self):
+        """
+        Ensure that the dimensions and coordinates are shared between cubes.
+
+        """
+        cubelist = self.cubelist
+        some_cube = self.cubelist[0]
+
+        for cube in cubelist:
+            for coord in cube.dim_coords:
+                assert coord.is_compatible(some_cube.coord(coord.name()))
+
+    def determine_dimensions(self, cube):
+        """
+        Determine the dimensions to collapse from the input cube.
+
+        """
+        dimensions = cube.dim_coords
+        dim_names = [dim.standard_name or dim.long_name for dim in dimensions]
+
+        if [self.cols] + [self.rows] in dim_names:
+            self.row.index = dim_names.index(self.row)
+
+    #@profile
+    def to_dataframe(self):
+        """
+        Turn the input cubes into a 2-dimensional DataFrame object
+
+        """
+
+        #coords   = [coord.standard_name or coord.long_name for coord in
+        #                                            self.cubelist[0].coords()]
+        #ignore = [item for item in coords if item not in [self.column_dims] +
+        #                                                 [self.primary_dim] +
+        #                                                 [self.pivot_dim]]
+        for cube in self.cubelist:
+            print cube
+            for coord in self.cubelist[0].coords():
+                name = coord.standard_name or coord.long_name
+
+                if (name not in self.column_dims + [self.primary_dim] +
+                                                   [self.pivot_dim]):
+                    cube.remove_coord(coord)
```

Comment:
What's the point of removing coordinate? The dimensions are still there if it is a dimensional coordinate, it just isn't named any more (i.e. it is an anonymous dimension).

Reply:
I think these can be safely removed but we should test.

Reply:
Yes, these had the purpose of marking which coords wanted to be mapped to columns (removing those which do not want) though this is not necessary.